30-helm/pingfederate-upgrade/hooks/pf-upgrade.sh

```diff
@@ -1,110 +1,160 @@
 fi
 
-while ! test -z ${1} ; do
-  case "${1}" in
-    --devops-user|-u)
-      export PING_IDENTITY_DEVOPS_USER="${1}"
-      shift;;
-    --devops-key|-k)
-      export PING_IDENTITY_DEVOPS_KEY="${1}"
-      shift;;
-    --verbose|-v)
-      set -x
-      shift;;
-    -h|--help)
-      exit_usage ;;
-    *)
-      exit_usage "Unrecognized Option ${1}" ;;
-  esac
-  shift
+# Read in passed options
+while ! test -z "${1}"; do
+    case "${1}" in
+        -u | --devops-user)
+            test -z "${2}" && usage "You must specify a Devops User with the ${1} option"
+            shift
+            export PING_IDENTITY_DEVOPS_USER="${1}"
+            ;;
+        -k | --devops-key)
+            test -z "${2}" && usage "You must specify a Devops Key with the ${1} option"
+            shift
+            export PING_IDENTITY_DEVOPS_KEY="${1}"
+            ;;
+        -h | --help)
+            usage
+            ;;
+        *)
+            usage "ERROR: Unrecognized Option ${1}"
+            ;;
+    esac
+    shift
 done
 
-if test -z "${NEW_PF_VERSION}" || test -z "${PING_IDENTITY_DEVOPS_USER}" || test -z "${PING_IDENTITY_DEVOPS_KEY}" ; then
-  exit_usage
+# Make sure all required variables are now present in the environment
+if test -z "${NEW_PF_VERSION}" || test -z "${PING_IDENTITY_DEVOPS_USER}" || test -z "${PING_IDENTITY_DEVOPS_KEY}"; then
+    usage "ERROR: One or more variables are not specified: PingFederate Version, Devops User, and Devops Key"
 fi
 
-set -e
-NEW_PF_VERSION="${NEW_PF_VERSION:-10.3.4}"
-
-sh /opt/staging/hooks/get-bits.sh -p pingfederate -v "${NEW_PF_VERSION}" -u "${PING_IDENTITY_DEVOPS_USER}" -k "${PING_IDENTITY_DEVOPS_KEY}" -c
-unzip -d /opt/new "/tmp/pingfederate-${NEW_PF_VERSION}.zip"
-  ## download license to /tmp/pingfederate.lic
-sh /opt/staging/hooks/get-bits.sh -p pingfederate -v "${NEW_PF_VERSION}" -u "${PING_IDENTITY_DEVOPS_USER}" -k "${PING_IDENTITY_DEVOPS_KEY}" -l -c
-
-_pfPort=$(netstat -tuln | grep 0.0.0.0:[0-9] | awk  '{print $4}' | cut -b9-)
-if test $(curl -kSs "https://127.0.0.1:${_pfPort}" --write-out '%{http_code}' 2>&1) -eq 302 -o 200 \
-  || test $(curl -kSs "http://127.0.0.1:${_pfPort}" --write-out '%{http_code}' 2>&1) -eq 302 -o 200; then 
-  echo "PF is running, stopping now"
-  pkill /opt/java/bin/java
-  ## let pf shutdown
-  sleep 15
+# Make sure "pingfederate-${NEW_PF_VERSION}.zip" has been provided at "/tmp/pingfederate-${NEW_PF_VERSION}.zip"
+echo "INFO: Checking for new PingFederate zip"
+new_pf_zip_path="/tmp/pingfederate-${NEW_PF_VERSION}.zip"
+if ! test -f "${new_pf_zip_path}"; then
+    echo "ERROR: No Pingfederate zip file found at file location ${new_pf_zip_path}"
+    exit 1
+fi
+
+# Extract new pingfederate bits
+echo "INFO: Extracting new PingFederate from zip"
+unzip -qd /opt/new "${new_pf_zip_path}"
+test "${?}" -ne 0 && echo "ERROR: Failed to unzip file ${new_pf_zip_path}" && exit 1
+
+# Make sure "pingfederate.lic" has been provided at "/tmp/pingfederate.lic
```

Comment:
Is a new license needed for a minor version upgrade? Like 11.0 to 11.1?

Reply:
I'm not sure this is needed on patch upgrades.

For minor upgrades, the license does specify major.minor version, so I would assume this is needed there. @samir-gandhi Thoughts?

$ pingctl license pingfederate 11.1
ID=00392207
WSTrustSTS=true
OAuth=true
SaasProvisioning=true
Product=PingFederate
Version=11.1
...

Reply:
apparently there are scenarios where pf may care about minor. so, better safe than sorry.